tests/filter_ntfs.rs

```diff
@@ -41,37 +39,50 @@ fn test_remove_network() {
     test_in_tmp_dir(|path| {
         let input_dir = "tests/fixtures/filter_ntfs/input";
 
-        let model = filter::filter(
-            transit_model::ntfs::read(input_dir).unwrap(),
-            filter::Action::Remove,
-            vec!["network1".into()],
-        )
-        .unwrap();
+        let mut filter = filter::Filter::new(filter::Action::Remove);
+        filter.add(filter::ObjectType::Network, "network_id", "network1");
+
+        let model = filter::filter(transit_model::ntfs::read(input_dir).unwrap(), &filter).unwrap();
         transit_model::ntfs::write(&model, path, get_test_datetime()).unwrap();
         compare_output_dir_with_expected(&path, None, "./tests/fixtures/filter_ntfs/output_remove");
     });
 }
 
 #[test]
-#[should_panic(expected = "network unknown not found.")]
+#[should_panic(expected = "Network \\'unknown\\' not found.")]
 fn test_extract_with_unknown_network() {
     let input_dir = "tests/fixtures/filter_ntfs/input";
-    filter::filter(
-        transit_model::ntfs::read(input_dir).unwrap(),
-        filter::Action::Extract,
-        vec!["unknown".into()],
-    )
-    .unwrap();
+    let mut filter = filter::Filter::new(filter::Action::Extract);
+    filter.add(filter::ObjectType::Network, "network_id", "unknown");
+
+    filter::filter(transit_model::ntfs::read(input_dir).unwrap(), &filter).unwrap();
 }
 
 #[test]
-#[should_panic(expected = "the data does not contain services anymore.")]
+#[should_panic(expected = "the data does not contain vehicle journeys anymore.")]
 fn test_remove_all_networks() {
     let input_dir = "tests/fixtures/filter_ntfs/input";
-    filter::filter(
-        transit_model::ntfs::read(input_dir).unwrap(),
-        filter::Action::Remove,
-        vec!["network1".into(), "network2".into(), "network3".into()],
-    )
-    .unwrap();
+    let mut filter = filter::Filter::new(filter::Action::Remove);
+    filter.add(filter::ObjectType::Network, "network_id", "network1");
+    filter.add(filter::ObjectType::Network, "network_id", "network2");
+    filter.add(filter::ObjectType::Network, "network_id", "network3");
+    filter::filter(transit_model::ntfs::read(input_dir).unwrap(), &filter).unwrap();
+}
+
+#[test]
+fn test_remove_line_by_line_code() {
```

Comment:
Can you a test ```test_extract_lines_by_line_code``` with route1 and route2 ?

Reply:
I added this test with `route1` and `route3`. See 0a8014bd7e7ba1b184cd45586bc3744e4b8dafa8.